addons/html_editor/static/src/main/link/link_selection_plugin.js

```diff
@@ -171,55 +147,56 @@ export class LinkSelectionPlugin extends Plugin {
-            !closestElement(node, "a") &&
-            // @todo review the need for "root" parameter
-            // This apparently depends on the selection state. Should it?
-            // Does it have to do with preserving cursor (avoid messing the selection)?
-            !(
-                closestElement(root, "[contenteditable=true]") &&
-                this.shared.getTraversedNodes().includes(node)
-            )
+    isLinkEligibleForVisualIndication(link) {
+        return (
+            this.isLinkEligibleForZwnbsp(link) &&
+            !this.resources.excludeLinkVisualIndication?.some((callback) => callback(link))
+        );
+    }
+
+    /**
+     * Apply the o_link_in_selection class if the selection is in a single link,
+     * remove it otherwise.
+     *
+     * @param {EditorSelection} [selection]
+     */
+    resetLinkInSelection(selection = this.shared.getEditableSelection()) {
+        this.clearLinkInSelectionClass(this.editable);
+
+        const { anchorNode, focusNode } = selection;
+        const [anchorLink, focusLink] = [anchorNode, focusNode].map((node) =>
+            closestElement(node, "a")
+        );
+        const singleLinkInSelection = anchorLink === focusLink && anchorLink;
+
+        if (
+            singleLinkInSelection &&
+            this.isLinkEligibleForVisualIndication(singleLinkInSelection)
         ) {
-            // Remove link ZWNBSP not in selection.
-            const startsWithLegitZws =
-                node.textContent.startsWith("\uFEFF") &&
-                node.previousSibling &&
-                node.previousSibling.nodeName === "A";
-            const endsWithLegitZws =
-                node.textContent.endsWith("\uFEFF") &&
-                node.nextSibling &&
-                node.nextSibling.nodeName === "A";
-            let newText = node.textContent.replace(/\uFEFF/g, "");
-            if (startsWithLegitZws) {
-                newText = "\uFEFF" + newText;
-            }
-            if (endsWithLegitZws) {
-                newText = newText + "\uFEFF";
-            }
-            if (newText !== node.textContent) {
-                // We replace the text node with a new text node with the
-                // update text rather than just changing the text content of
-                // the node because these two methods create different
-                // mutations and at least the tour system breaks if all we
-                // send here is a text content change.
-                const newTextNode = this.document.createTextNode(newText);
-                node.before(newTextNode);
-                node.remove();
-                node = newTextNode;
+            singleLinkInSelection.classList.add("o_link_in_selection");
+        }
+    }
+
+    clearLinkInSelectionClass(root) {
+        for (const link of selectElements(root, ".o_link_in_selection")) {
+            removeClass(link, "o_link_in_selection");
+        }
+    }
+
+    removeEmptyLinks(root) {
```

Comment:
If you want, I think we can move this function to the link plugin cause it cleans up the empty links and feels more related there

Reply:
You're right, it should be in the Link Plugin. I'll move it.

Reply:
Some other stuff should also be in the LinkPlugin, like handling Enter and Shift+Enter at the edges of links. I'm updating the PR.